crates/fluvio-dataplane-protocol/src/produce/request.rs

```diff
@@ -91,6 +79,99 @@ where
     pub records: R,
 }
 
+impl<R> Encoder for ProduceRequest<R>
```

Comment:
Instead of writing custom encoder, you should directly implement encoder and decoder for duration in fluvio protocol crate

Reply:
This encoding of duration and timeout is specific for produce requests only. For other places, it fairly might be done differently.

Reply:
For simplicity, I think we should provide a default implementation.  Could always provide custom implementation later if needs arise.